src/Forms/UnsavedChangesIndicator.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace SilverStripe\Forms;
+
+use SilverStripe\Forms\FormAction;
+
+/**
+ * A visual indicator to show whether a form has unsaved changes.
+ *
+ * Note that this isn't strictly a FormAction as it doesn't perform any action when clicked.
+ * However, extending FormAction allows us to easily integrate with the existing form action
+ * architecture (e.g. placement in the form actions area).
+ */
+class UnsavedChangesIndicator extends FormAction
+{
+    private static array $minutes = [
+        'notice' => 5,
+        'warning' => 10,
+    ];
```

Comment:
Why minutes instead of seconds? Doesn't really matter but I'd like to understand the reasoning if just so I can answer this question if someone asks me.

Reply:
It's more intuitive IMO

Also it encourages people to not put in fractions of minutes (which you could do as we're not validating the config at all), however on the front end it will decimal as a decimal e.g. 1.5 mins - the design did not include a seconds component, nor would I really want it to because the component already takes up a lot of horizontal real-estate which is particularly problematic at smaller screen widths.